lib/multicell_location/services/nrf_cloud_integration.c

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <modem/lte_lc.h>
+#include <cJSON.h>
+#include <cJSON_os.h>
+#include <modem/at_cmd.h>
+
+#include "location_service.h"
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(multicell_location_nrf_cloud, CONFIG_MULTICELL_LOCATION_LOG_LEVEL);
+
+#define HOSTNAME CONFIG_MULTICELL_LOCATION_HOSTNAME
+
+/* Estimated size requirements for HTTP header */
+#define HTTP_HEADER_SIZE	200
+/* Estimated size for current cell information */
+#define CURRENT_CELL_SIZE	100
+/* Estiamted size for HTTP request body */
+#define HTTP_BODY_SIZE		(CURRENT_CELL_SIZE + 200)
+/* Length of nrf9160 device IMEI */
+#define IMEI_LEN 15
+
+BUILD_ASSERT(sizeof(CONFIG_MULTICELL_LOCATION_SERVICE_NRF_CLOUD_API_KEY) > 1,
+	     "API key must be configured");
+#define API_KEY CONFIG_MULTICELL_LOCATION_SERVICE_NRF_CLOUD_API_KEY
+
+#define HTTP_REQUEST_HEADER						\
+	"POST /v1/location/single-cell HTTP/1.1\r\n"			\
+	"Host: "HOSTNAME"\r\n"						\
+	"Authorization: Bearer "API_KEY"\r\n"				\
+	"Content-Length: %d\r\n\r\n"
+
+#define HTTP_REQUEST_BODY "&deviceIdentifier=nrf-%s"			\
+			  "&mcc=%u"					\
+			  "&mnc=%u"					\
+			  "&tac=%u"					\
+			  "&eci=%u"					\
+			  "&format=json"
+
+BUILD_ASSERT(sizeof(HOSTNAME) > 1, "Hostname must be configured");
+
+static char body[HTTP_BODY_SIZE];
+
+static const char tls_certificate[] =
```

Comment:
Would be good to add some info on the certificate origin, same as for other providers.

Reply:
Yeah. Agree. @jtguggedal will continue with this PR.

Reply:
Certificate info has been added